components/admin/Users/ConfirmDeactivateModal.tsx

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import useSWR from "swr";
+import { useTranslation } from "next-i18next";
+import { FormElementTypes } from "@lib/types";
+import { Button } from "@components/globals";
+import { Dialog, useDialogRef } from "@components/form-builder/app/shared";
+import { updateActiveStatus } from "@pages/admin/accounts";
+import { useRefresh } from "@lib/hooks";
+import Loader from "@components/globals/Loader";
+import { Attention, AttentionTypes } from "@components/globals/Attention/Attention";
+import { DBUser } from "@lib/types/user-types";
+
+const fetcher = async (url: string) => {
+  const res = await fetch(url);
+
+  if (res.status === 200) {
+    const data = await res.json();
+    return data;
+  }
+
+  // handle using swr error
+  throw new Error("Something went wrong");
+};
+
+export const ConfirmDeactivateModal = ({
+  handleClose,
+  user,
+}: {
+  handleAddType?: (type?: FormElementTypes) => void;
+  handleClose: () => void;
+  user: DBUser;
+}) => {
+  const { t } = useTranslation("admin-users");
+  const dialog = useDialogRef();
+  const { refreshData } = useRefresh();
+
+  const { data, isLoading, error } = useSWR(`/api/account/${user.id}/forms`, fetcher);
```

Comment:
I'm wondering if the use of SWR (stale-while-revalidate) might cause unintended issues.
Thinking if a user loads the modal and get's a message that the user has existing forms.  The user changes ownership of forms in a different tab or by navigating where useSWR isn't called again.  This modal could potentially fetch stale data that a user still have forms while in the database fresh data has no forms associated.

Reply:
@dsamojlenko can you validate this one?

Reply:
![Screen Recording 2023-07-11 at 1 10 03 PM](https://github.com/cds-snc/platform-forms-client/assets/1187115/f27a0692-28a4-4757-a74e-102854cdf68b)

Does this cover the scenario you're describing?

Reply:
It does.  Interesting, I guess multiple renders are causing the data to update.

Reply:
@dsamojlenko is that running in Dev mode ? 

Just want to confirm same for a prod version.

Reply:
Works same in PR Review environment

Reply:
I think I know what's going on, but it doesn't matter we can leave as is.
If you're curious though...
The tab switching is causing the Next Auth session context to update which rerenders all children including the modal.  Next Auth updates that Session Context Provider when it detects tab switching to ensure the auth status of a user is consistent across tabs.